src/main/java/ru/hogwarts/school/controller/FacultyController.java

```diff
@@ -0,0 +1,67 @@
+package ru.hogwarts.school.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import org.springframework.web.bind.annotation.*;
+import ru.hogwarts.school.entity.Faculty;
+import ru.hogwarts.school.entity.Student;
+import ru.hogwarts.school.service.FacultyService;
+
+import java.util.List;
+import java.util.Set;
+
+@RestController
+@RequestMapping("/faculty")
+public class FacultyController {
+    private final FacultyService facultyService;
+
+    public FacultyController(FacultyService facultyService) {
+        this.facultyService = facultyService;
+    }
+
+    @PostMapping
+    public Faculty postFaculty(@RequestBody Faculty faculty) {
+        return facultyService.postFaculty(faculty);
+    }
+
+    @GetMapping
+    @Operation(summary = "Get all faculties")
+    public List<Faculty> findAll() {
+        return facultyService.findAll();
+    }
+
+    @GetMapping(value = "/color", params = "color")
+    @Operation(summary = "Get faculties by color")
+    public List<Faculty> getFacultiesByColor(@RequestParam String color) {
+        return facultyService.findByColor(color);
+    }
+
+    @GetMapping(value = "/nameOrColor", params = "nameOrColor")
```

Comment:
```suggestion
    @GetMapping(params = "nameOrColor")
```
снова та же ошибка - в пути ничего писать не нужно здесь